src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let state1 = { ...state };
+  const arr = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      state1 = { ...state1, ...obj.extraData };
+      arr.push({ ...state1 });
+    } else if (obj.type === 'removeProperties') {
+      for (const key of obj.keysToRemove) {
+        delete state1[key];
+      }
+      arr.push({ ...state1 });
+    } else if (obj.type === 'clear') {
+      state1 = {};
+      arr.push(state1);
```

Comment:
[DONT REPEAT YOURSELF] - Pushing a copy of the object to the array is done in every case of the action type. To avoid repeating this action, do it just once after the switch statement.